text/0000-group-by.md

```diff
@@ -0,0 +1,101 @@
+- Feature Name: group_by
+- Start Date: 2018-06-15
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+Provide an `Iterator` over a slice that produce non-overlapping runs of elements separated by a given predicate.
+
+# Motivation
+[motivation]: #motivation
+
+Adding this `Iterator` to the standard library will help people split slices by using a custom predicate!
+This `Iterator` is implemented on generic slices to provide performances and flexibility, `GroupBy` implements `DoubleEndedIterator` without any overhead and it does not need any allocation.
+
+There is a similar method that already exists in [the standard library called `split`](https://doc.rust-lang.org/std/primitive.slice.html#method.split) but it will remove the element that does the separation.
```

Comment:
You still haven’t explained *why* this should be added to `std` / `core` instead of `itertools`. If `itertools` must be fixed or augmented with a new function, this function belongs to `itertools`, not `std`.

Reply:
I updated the motivation section.